TheWarWithin/Priorities/DeathKnightFrost.simc

```diff
@@ -122,61 +148,33 @@ actions.obliteration+=/horn_of_winter,if=rune<3
 actions.obliteration+=/arcane_torrent,if=rune<1&runic_power<30
 actions.obliteration+=/howling_blast,if=!buff.killing_machine.up
 
-# Racial Abilities
-actions.racials+=/blood_fury,if=variable.cooldown_check
-actions.racials+=/berserking,if=variable.cooldown_check
-actions.racials+=/arcane_pulse,if=variable.cooldown_check
-actions.racials+=/lights_judgment,if=variable.cooldown_check
-actions.racials+=/ancestral_call,if=variable.cooldown_check
-actions.racials+=/fireblood,if=variable.cooldown_check
-actions.racials+=/bag_of_tricks,if=talent.obliteration&!buff.pillar_of_frost.up&buff.unholy_strength.up
-actions.racials+=/bag_of_tricks,if=!talent.obliteration&buff.pillar_of_frost.up&(buff.unholy_strength.up&buff.unholy_strength.remains<gcd*3|buff.pillar_of_frost.remains<gcd*3)
+
+# AoE Action List
+actions.aoe+=/obliterate,if=buff.killing_machine.up&talent.cleaving_strikes&buff.death_and_decay.up
+actions.aoe+=/frost_strike,cycle_targets=1,if=!variable.pooling_runic_power&debuff.razorice.stack=5&talent.shattering_blade&(talent.shattered_frost|active_enemies<4)
+actions.aoe+=/howling_blast,cycle_targets=1,if=!dot.frost_fever.ticking
+actions.aoe+=/howling_blast,if=buff.rime.up
+actions.aoe+=/obliterate,if=buff.killing_machine.stack>0
+actions.aoe+=/glacial_advance,cycle_targets=1,if=!variable.pooling_runic_power&(variable.ga_priority|debuff.razorice.stack<5)
+actions.aoe+=/frost_strike,cycle_targets=1,if=!variable.pooling_runic_power
+actions.aoe+=/obliterate
+actions.aoe+=/horn_of_winter,if=rune<2&runic_power.deficit>25&(!talent.breath_of_sindragosa|variable.true_breath_cooldown>cooldown.horn_of_winter.duration-15)
+actions.aoe+=/arcane_torrent,if=runic_power.deficit>25
+
 
 # Single Target Rotation
 actions.single_target+=/frost_strike,if=talent.a_feast_of_souls&debuff.razorice.stack=5&talent.shattering_blade&buff.a_feast_of_souls.up
 actions.single_target+=/obliterate,if=buff.killing_machine.stack=2|buff.exterminate.up
-actions.single_target+=/horn_of_winter,if=(!talent.breath_of_sindragosa|variable.true_breath_cooldown>cooldown.horn_of_winter.duration-15)&cooldown.pillar_of_frost.remains<variable.oblit_pooling_time
 actions.single_target+=/frost_strike,if=(debuff.razorice.stack=5&talent.shattering_blade)|(rune<2&!talent.icebreaker)
-actions.single_target+=/howling_blast,if=variable.rime_buffs
+actions.single_target+=/howling_blast,if=variable.rime_buffs&talent.icebreaker
 actions.single_target+=/obliterate,if=buff.killing_machine.up&!variable.pooling_runes
-actions.single_target+=/glacial_advance,if=!variable.pooling_runic_power&!death_knight.runeforge.razorice&(debuff.razorice.stack<5|debuff.razorice.remains<gcd*3)
+actions.single_target+=/soul_reaper,if=fight_remains>5&target.time_to_pct_35<5&target.time_to_pct_0>5&rune>2
```

Comment:
Probably `boss&fight_remains` on this one? Most of the others are fine because they're indirectly linked to being in a cooldown window already.

Reply:
There is already a fight remains on it, it's the first conditional in the statement. Doesn't really matter if it's a boss or not.

Reply:
Yeah, `fight_remains>5` is to ensure Soul Reaper has time to pop.  This isn't a "use Soul Reaper at end of combat" entry.